src/olmo_core/data/source_mixture.py

```diff
@@ -287,7 +287,6 @@ def validate(self):
 
     def build(self, *, npdtype: NumpyUIntTypes, sequence_length: int) -> SourceMixtureDataset:
         self.validate()
-        random.seed(self.seed)
```

Comment:
We set all random seeds at the beginning of the job. Additionally, we construct RNG generators directly when sampling instances anyways.